### DISCUSSION TITLE
Apache and related containers can not start up correctlly. (`NOTICE: PHP message: Slim Application Error`)

### DISCUSSION THREAD

**Discussion by DF-wu** [nextcloud/all-in-one#3552] [Questions] [answered]
update in reply

---
Issue descriptioin: 
I would like to start a fresh new nextcloud AIO instead of my old nextcloud container.  Follow the document and set-up.  But some aio containers can not start up. The log message shows `NOTICE: PHP message: Slim Application Error`.

I have read lots of related post but still can not figure out why it happen. 
If more log is needed, please tell me how to produce it.
please help me troublshooting.  And big thanks for this project.

My Enviroments:
+ Portainer-CE with official docker compose deployment.
+ AIO is behind reverse proxy (Nginx Proxy Manager) so of couse I read all the doc and follow step-by-step.
+ System Information:
    + [docker_info.txt](https://github.com/nextcloud/all-in-one/files/13010915/docker_info.txt)
    + neofetch of my os ![](https://github.com/nextcloud/all-in-one/assets/28750726/8cb8962a-49c3-4102-8af2-86ff5299da69)
+ my deployment docker-compose.yml
    + [link to my github repo](https://github.com/DF-wu/myServices/blob/bfd2e3151ac8578b23da23841fbe10b8115916c6/nextcloud-aio/docker-compose.yml)

Related logs or screen shot:
+ The mastercontainer_logs: [_nextcloud-aio-mastercontainer_logs (1).txt](https://github.com/nextcloud/all-in-one/files/13010287/_nextcloud-aio-mastercontainer_logs.1.txt)

+ The screen shot to show the container situation : 
    + ![image](https://github.com/nextcloud/all-in-one/assets/28750726/6e98f744-143c-4ca3-840a-4dbaf7faa283)

My troubleshooting:
+ Try launch it either Portainer or Bash.
+ Following https://github.com/nextcloud/all-in-one#how-to-properly-reset-the-instance every time I stop the container.
+ Configure my NginxProxyManager with different image and setting. Including reset it.
+ Search related message like my situation. Unfortunately,  problem is still there.
+ Use another vm as test enviroment. Same issue.
+ I execute the command which is frequently mentioned in similar posts.
    + `sudo docker exec -it nextcloud-aio-mastercontainer docker pull nextcloud/aio-collabora:latest
`
    + and the result like this
    + ![image](https://github.com/nextcloud/all-in-one/assets/28750726/ebb4b144-563a-4858-bcee-31e0828d015b)

**Comment by DF-wu** [comment, marked as answer]
updata 2:
TL;DR  NFS not working in my situation.

I have tried different of mounting option either`nextcloud_aio_nextcloud_datadir` nor host path. Seems like the NFS just not work.
[The error message is like this. ](https://github.com/nextcloud/all-in-one/files/13038355/_nextcloud-aio-mastercontainer_logs.2.txt)

At the end, I switch to CIFS with `autoFS` and it works. [The setting is simply like this.](https://github.com/DF-wu/myServices/commit/5a147143ff3f8319f282828b092494b2a92dcd8c#diff-7cf290eb3ad113f6973a2fbccb2cbe461ff55d109759e4d6e039d5fa6783d7f7R22)  
I will keep following up. 

I would like to know if you guys can re-produce the issue.
If this issue is confirmed, maybe I can give some note in doc?


---

update 1:  

I think I found the critical factor for this case.
In my enviroment, the `/mnt/mydata` is actually an NFS mounting point which is controlled by `autfofs.service`. 
After I change the value of `NEXTCLOUD_DATADIR` from `/mnt/mydata/ncdata` to `/home/df/ncdata_test` (a regular directory in host). These containers is launched properlly even the logs still has some error message.
As a storage concern, I will try mount docker volume with NFS and softlink to my desire path.
